Packages/doc/databrowser_paplot.rst

```diff
@@ -0,0 +1,223 @@
+
+Image plot
+""""""""""
+
+.. figure:: PulseAverage_images.png
+   :scale: 50%
+   :figwidth: 70%
+   :align: left
+
+.. figure:: PulseAverage_images_colorscales.png
+   :height: 870px
+   :align: right
+
+.. raw:: html
+
+   <p>&nbsp;&nbsp;&nbsp;<br></p>
+
+This graph is an alternative display to the :ref:`Figure Pulse Average Trace
+plot`. Its advantage is a much faster updates for displaying many overlayed
+sweeps. Each set image has the average and the deconvolution data at the
+bottom. When these are not shown a blank space is left empty.
+On top of these the single pulse data is shown. Each data is displayed as rows
+(horizontally), and the image is filled from bottom to top depending on the
+:ref:`db_paplot_image_sortorder`.
+
+.. _db_paplot_image_sortorder:
+
+Sort Order
+^^^^^^^^^^
+
+The following tables visualizes the display of one image set with two sweeps
+overlayed and three pulses using different ``Sort Order`` settings. The
+ordering is always ascending and from bottom to top. Due to implementation
+details the ``Sweep`` sort order allows **much** faster incremental updates
+(only relevant during data acqisition).
+
+.. table:: Comparing different sort orders
+
+   +----------------+----------------+
+   | Sweep          | Pulse          |
+   +================+================+
+   |Pulse 2, Sweep 1|Pulse 2, Sweep 1|
+   +----------------+----------------+
+   |Pulse 1, Sweep 1|Pulse 2, Sweep 0|
+   +----------------+----------------+
+   |Pulse 0, Sweep 1|Pulse 1, Sweep 1|
+   +----------------+----------------+
+   |Pulse 2, Sweep 0|Pulse 1, Sweep 0|
+   +----------------+----------------+
+   |Pulse 1, Sweep 0|Pulse 0, Sweep 1|
+   +----------------+----------------+
+   |Pulse 0, Sweep 0|Pulse 0, Sweep 0|
+   +----------------+----------------+
+   |Deconvolution   |Deconvolution   |
+   +----------------+----------------+
+   |Average         |Average         |
+   +----------------+----------------+
+
+.. _db_paplot_timealignment:
+
+Time Alignment
+^^^^^^^^^^^^^^
+
+The pulses from different channels might not have the same time basis. Time alignment is a way to correct for these offsets.
+
+The algorithm is as follows:
+
+- Get the feature position ``featurePos`` for all pulses which belong to the
+  same set. Store these feature positions using their sweep number and pulse
+  index as key.
+- Now shift all pulses in all sets from the same region by ``-featurePos``
+  where ``featurePos`` is used from the same sweep and pulse index.
+
+.. _db_paplot_operation_order:
+
+Operation order
+^^^^^^^^^^^^^^^
+
+The operations on the data happens in the following fixed order:
+
+- Gather pulses
+- Apply pulse sort order
+- Failed pulse searching
+- Zeroing
+- Time alignment
+- Averaging
+
+PA plot settings
+^^^^^^^^^^^^^^^^
+
+The following figure shows the browser settings panel for the PA plot. It can
+be reached from the ``PA`` tab in the Databrowser. The next sections explain each control.
```

Comment:
The following sections describe PA settings. Settings are configured on the data browser PA settings tab (shown below).